test/sig_utils.py

```diff
@@ -301,3 +312,365 @@ def reference_vector_argument(arg):
     if arg in ("array[] real", "row_vector"):
         return "vector"
     return arg
+
+overload_scalar = {
+    "Prim": "double",
+    "PrimExp": "double",
+    "Rev": "stan::math::var",
+    "RevSOA": "stan::math::var",
+    "RevExp": "stan::math::var",
+    "Fwd": "stan::math::fvar<double>",
+    "FwdExp": "stan::math::fvar<double>",
+    "Mix": "stan::math::fvar<stan::math::var>",
+}
+
+class CppStatement:
+    def init(self):
+        raise Exception("CppStatement should never be instantiated directly")
+
+    def is_reverse_mode(self):
+        return False
+    
+    def is_matrix_like(self):
+        return False
+    
+    def is_varmat(self):
+        return False
+    
+    def is_expression(self):
+        return False
```

Comment:
These methods seem kinda redundand in statements. I know they are used in `FunctionGenerator`, but I think it would be simpler to implement them there.

Reply:
These are inherited by all the different argument types. I was doing a polymorphism to make the code shorter.

Reply:
I understand. But as I see it we don't need it. `is_varmat` and `is_expression` seem to be unused, calls `is_reverse_mode` on arguments can be replaced by a check whether we are doing a reverse mode overload and calls to `is_matrix_like` can be replaced with a check is something is eigen type.